python/GafferUI/VectorDataWidget.py

```diff
@@ -76,11 +76,16 @@ def __init__(
 		data=None,
 		editable=True,
 		header=False,
+		headerPrefix=None,
 		showIndices=True,
 		minimumVisibleRows=8,
 		columnToolTips=None,
 		sizeEditable=True,
 		columnEditability=None,
+		horizontalScrollMode = GafferUI.Enums.ScrollMode.Never,
+		verticalScrollMode = GafferUI.Enums.ScrollMode.Automatic,
+		overrideStretch = False,
+		extendColumnToolTipsToData = True,
```

Comment:
I'm concerned that this widget is becoming a bit too much of a beast. There seem to be three of four separate changes here on top of the "better support for different data types". Could you split this commit into its constituent parts, with an explanation of the need for each one? I want to understand what we're changing and why, and any potential knock-on effects to existing code, and I'm struggling a bit with it all in one change...